java-manta-client-unshaded/src/main/java/com/joyent/manta/client/PruneEmptyParentDirectoryStrategy.java

```diff
@@ -73,13 +105,45 @@ static void pruneParentDirectories(final MantaClient client,
 
                 client.delete(directories[i], headers, null);
             } catch (MantaClientHttpResponseException responseException) {
-                if (responseException.getServerCode().equals(MantaErrorCode.RESOURCE_NOT_FOUND_ERROR)) {
-                    continue;
-                } else if (responseException.getServerCode() != MantaErrorCode.DIRECTORY_NOT_EMPTY_ERROR) {
-                    throw responseException;
+                switch (responseException.getServerCode()) {
+                    /* The directory to be deleted was deleted by someone else
+                     * so we stop pruning. */
+                    case RESOURCE_NOT_FOUND_ERROR:
+                        LOG.debug("Directory not found: {}", directories[i]);
+                        return;
+                    /* The directory to be deleted was not empty so we stop
+                       pruning. */
+                    case DIRECTORY_NOT_EMPTY_ERROR:
+                        LOG.debug("Directory not empty: {}", directories[i]);
+                        return;
+                    default:
+                        throw responseException;
                 }
-
             }
         }
     }
+
+    /**
+     * Method that will product an array of strings representing each directory
+     * in a path ordered from right to left with the deepest object excluded.
+     * Additionally, immutable base paths like /user/stor will be excluded
+     * from the array of directories.
+     *
+     * @param path path to extract directories from
+     * @return array of directory full paths
+     */
+    static String[] parseDirectoriesFromPathRightToLeft(final String path) {
```

Comment:
I would call this method getOrderedWriteableAncestorPaths and then call out the order in the method comment.

Reply:
See above.